tests/components/picnic/test_sensor.py

```diff
@@ -0,0 +1,407 @@
+        "street": "Commonstreet",
+        "city": "Somewhere",
+    },
+    "total_deliveries": 123,
+    "completed_deliveries": 112,
+}
+DEFAULT_CART_RESPONSE = {
+    "items": [],
+    "delivery_slots": [
+        {
+            "slot_id": "611a3b074872b23576bef456a",
+            "window_start": "2021-03-03T14:45:00.000+01:00",
+            "window_end": "2021-03-03T15:45:00.000+01:00",
+            "cut_off_time": "2021-03-02T22:00:00.000+01:00",
+            "minimum_order_value": 3500,
+        },
+    ],
+    "selected_slot": {"slot_id": "611a3b074872b23576bef456a", "state": "EXPLICIT"},
+    "total_count": 10,
+    "total_price": 2535,
+}
+DEFAULT_DELIVERY_RESPONSE = {
+    "delivery_id": "z28fjso23e",
+    "creation_time": "2021-02-24T21:48:46.395+01:00",
+    "slot": {
+        "slot_id": "602473859a40dc24c6b65879",
+        "hub_id": "AMS",
+        "window_start": "2021-02-26T20:15:00.000+01:00",
+        "window_end": "2021-02-26T21:15:00.000+01:00",
+        "cut_off_time": "2021-02-25T22:00:00.000+01:00",
+        "minimum_order_value": 3500,
+    },
+    "eta2": {
+        "start": "2021-02-26T20:54:00.000+01:00",
+        "end": "2021-02-26T21:14:00.000+01:00",
+    },
+    "status": "COMPLETED",
+    "delivery_time": {
+        "start": "2021-02-26T20:54:05.221+01:00",
+        "end": "2021-02-26T20:58:31.802+01:00",
+    },
+    "orders": [
+        {
+            "creation_time": "2021-02-24T21:48:46.418+01:00",
+            "total_price": 3597,
+        },
+        {
+            "creation_time": "2021-02-25T17:10:26.816+01:00",
+            "total_price": 536,
+        },
+    ],
+}
+
+
+@pytest.mark.usefixtures("hass_storage")
+class TestPicnicSensor(unittest.IsolatedAsyncioTestCase):
+    """Test the Picnic sensor."""
+
+    async def asyncSetUp(self):
+        """Set up things to be run when tests are started."""
+        self.hass = await async_test_home_assistant(None)
+        self.entity_registry = (
+            await self.hass.helpers.entity_registry.async_get_registry()
+        )
+
+        # Patch the api client
+        self.picnic_patcher = patch("homeassistant.components.picnic.PicnicAPI")
+        self.picnic_mock = self.picnic_patcher.start()
+
+        # Add a config entry and setup the integration
+        config_data = {
+            CONF_ACCESS_TOKEN: "x-original-picnic-auth-token",
+            CONF_COUNTRY_CODE: "NL",
+        }
+        self.config_entry = MockConfigEntry(
+            domain=const.DOMAIN,
+            data=config_data,
+            connection_class=CONN_CLASS_CLOUD_POLL,
+            unique_id="295-6y3-1nf4",
+        )
+        self.config_entry.add_to_hass(self.hass)
+
+    async def asyncTearDown(self):
+        """Tear down the test setup, stop hass/patchers."""
+        await self.hass.async_stop(force=True)
+        self.picnic_patcher.stop()
+
+    @property
+    def _coordinator(self):
+        return self.hass.data[const.DOMAIN][self.config_entry.entry_id][
```

Comment:
Please don't interact with integration details in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations